src/dune_cache_storage/layout.ml

```diff
@@ -1,25 +1,50 @@
 open Stdune
 open Import
 
-let default_root_dir =
+let ( / ) = Path.relative
+
+(** The default directory of all caches. Defaults to [$XDG_CACHE_HOME/dune].
+    Not to be used directly, add a layer beforehand. *)
+let default_cache_dir =
   lazy
     (let cache_dir = Xdg.cache_dir (Lazy.force Dune_util.xdg) in
-     Path.L.relative (Path.of_filename_relative_to_initial_cwd cache_dir) [ "dune"; "db" ])
+     Path.of_filename_relative_to_initial_cwd cache_dir / "dune")
+;;
+
+let check_absolute var path =
+  if Filename.is_relative path
+  then
+    User_error.raise
+      [ Pp.paragraphf "$%s should be an absolute path, but is %S" var path ]
 ;;
 
-let root_dir =
+(** The home directory for all the caches. Defaults to [default_cache_dir],
+    or uses [$DUNE_CACHE_HOME] if set. *)
+let home_dir =
+  lazy
+    (let var = "DUNE_CACHE_HOME" in
+     match Sys.getenv_opt var with
+     | None -> Lazy.force default_cache_dir
+     | Some path ->
+       check_absolute var path;
+       Path.external_ (Path.External.of_string path))
```

Comment:
Genuine question: How do we decide it better to force `default_cache_dir` here rather than to return the other branch as lazy?